plugins/tiddlywiki/multiwikiserver/modules/commands/mws-add-user.js

```diff
@@ -0,0 +1,58 @@
+/*\
+title: $:/plugins/tiddlywiki/multiwikiserver/commands/mws-add-user.js
+type: application/javascript
+module-type: command
+
+Command to create users and grant permission
+
+\*/
+(function(){
+
+  /*jslint node: true, browser: true */
+  /*global $tw: false */
+  "use strict";
+  
+  var crypto = require("crypto");
+  
+  exports.info = {
+    name: "mws-add-user",
+    synchronous: false
+  };
+  
+  var Command = function(params,commander,callback) {
+    this.params = params;
+    this.commander = commander;
+    this.callback = callback;
+  };
+  
+  Command.prototype.execute = function() {
+    var self = this;
+  
+    if(this.params.length < 2) {
+      return "Usage: --mws-add-user <username> <password> [email]";
+    }
+  
+    if(!$tw.mws || !$tw.mws.store || !$tw.mws.store.sqlTiddlerDatabase) {
```

Comment:
I think the usernames and passwords should never be stored with the content. So for me `!$tw.mws.store.sqlTiddlerDatabase` looks like it is in the same database file?

Reply:
Hi @pmario I think it's perfectly OK for this data to be in the same database. There's no more obvious way for queries of the tiddler database to accidentally start returning results from the user database than there is for any data corruption within a program.

What I do want to pay attention to is the best practice of making the admin UI run off of a different HTTP host/port, so that administrators can use external tools to lock down admin access.

Reply:
The plan is to have pluggable authentication schemes. Form-based authentication is still pretty standard for website logins and so I don't think we can avoid offering it, but I definitely want to support things like logging in with Google or GitHub credentials.